decision/neighbour_filtering.py

```diff
@@ -0,0 +1,60 @@
+from globals import BATTERY_LIMIT
+import bisect
+import numpy as np
+
+
+def filtering_neighbours(state, number_of_neighbours=3, random_neighbours=0):
+    """
+    Filtering out neighbours based on a score of deviation of ideal state and distance from current cluster
+    :param state: state object to evaluate
+    :param number_of_neighbours: number of neighbours to be returned
+    :param random_neighbours: int - number of random neighbours to be added to the neighbourhood list
+    :return:
+    """
+    clusters = state.clusters
+    distance_to_all_clusters = state.get_distance_to_all(state.current_cluster.id)
+    max_dist, min_dist = max(distance_to_all_clusters), min(distance_to_all_clusters)
+    distance_scores = [
+        (dist - min_dist) / (max_dist - min_dist) for dist in distance_to_all_clusters
+    ]
+
+    deviation_ideal_states = [
+        abs(cluster.ideal_state - len(cluster.get_valid_scooters(BATTERY_LIMIT)))
+        for cluster in clusters
+    ]
+
+    max_deviation, min_deviation = (
+        max(deviation_ideal_states),
+        min(deviation_ideal_states),
+    )
+
+    deviation_scores = [
+        1 - ((deviation - min_deviation) / (max_deviation - min_deviation))
+        for deviation in deviation_ideal_states
+    ]
```

Comment:
How should we handle divide by zero here? Got this error message when running the tests locally
```
Traceback (most recent call last):
  File "/Users/sverrespetalen/KODE/master-thesis/env/lib/python3.7/unittest/case.py", line 59, in testPartExecutor
    yield
  File "/Users/sverrespetalen/KODE/master-thesis/env/lib/python3.7/unittest/case.py", line 628, in run
    testMethod()
  File "/Users/sverrespetalen/KODE/master-thesis/decision/test.py", line 122, in test_deliveries
    actions = self.initial_state.get_possible_actions()
  File "/Users/sverrespetalen/KODE/master-thesis/classes/State.py", line 132, in get_possible_actions
    random_neighbours=random_neighbours,
  File "/Users/sverrespetalen/KODE/master-thesis/decision/neighbour_filtering.py", line 33, in filtering_neighbours
    for deviation in deviation_ideal_states
  File "/Users/sverrespetalen/KODE/master-thesis/decision/neighbour_filtering.py", line 33, in <listcomp>
    for deviation in deviation_ideal_states
ZeroDivisionError: division by zero
```
A cluster can be in the ideal state from the beginning